app/controllers/gift_requests_controller.rb

```diff
@@ -7,20 +7,24 @@ class GiftRequestsController < ApplicationController
 
   def new
     @back_route = wishlist_path(params[:id])
-
     @gift_request = GiftRequest.new
-    render "gift_request", locals: {list_id: params[:id]}
+    @list_name = List.find(params[:id]).name
+
+    render "gift_request"
   end
 
   def edit
     @back_route = wishlist_path(params[:id])
-
-    render "gift_request", locals: {list_id: params[:id]}
+    @list_name = List.find(params[:id]).name
```

Comment:
You can just get the list name from `@gift_request` in `edit` instead of doing a separate query.